### DISCUSSION TITLE
New 0.71" Round Display

### DISCUSSION THREAD

**Discussion by Basitadam** [Bodmer/TFT_eSPI#3520] [Compatible displays and setup files]
Hi,
Anyone played with this little thing: https://tr.aliexpress.com/item/1005007531935475.html
I tried with these attached files and works quite good. Only there is no backlight pin, it is always on
[GC9D01_Defines.txt](https://github.com/user-attachments/files/17472786/GC9D01_Defines.txt)
[GC9D01_Init.txt](https://github.com/user-attachments/files/17472787/GC9D01_Init.txt)
[GC9D01_Rotation.txt](https://github.com/user-attachments/files/17472788/GC9D01_Rotation.txt)

Of course you need to change file extensions to .h put them in TFT_Drivers folder and make modifications to TFT_eSPI.cpp and TFT_eSPI.h

**Comment by Basitadam** [comment]
Sample images:

![071_1](https://github.com/user-attachments/assets/35be809b-91ee-4bdd-9716-fd8613abfc69)
![071_2](https://github.com/user-attachments/assets/0eb80731-c5ae-4c9a-b779-5f8c368f965d)

**Comment by webfraggle** [comment]
Hello @Basitadam ,

Thank you for your work. 
That takes me a step further. I have a 0.99’ display with 40x160 pixels and unfortunately it looks like the photo.
Can you share your changes to the other files?
Do you know what the problem might be?
Thank you very much
Christoph
![Groß (IMG_9332)](https://github.com/user-attachments/assets/c0c863a7-40d7-40f9-b067-8b7eb3b77d3b)

**Comment by Basitadam** [comment]
Hi Christoph,
Thank you.

For 0.71" you can download all the files via this link: https://www.filemail.com/d/imvnzqfhncrrcue
Please make sure to backup your original files first. Just in case. Been some time, I might have missed a file or modification to send, let me know if it doesnt work.

For 0.99" which TFT controller is used? Seems like scan issue
Tim

**Reply by jaredism0** [reply]
Could you please provide the files again? the link has expired. I'm using the ESP32S3 with the waveshare .71" display, and the most ive gotten beyond backlight is inversion and changes in brightness/white lines on the lcd.

Thank you

**Reply by Basitadam** [reply]
Hi,
Here is the link : https://www.filemail.com/d/rkylhockysjcxew

**Comment by webfraggle** [comment]
Hello Tim,
thank you for your files. Just tried it, but the result is the same. I'm using this display https://de.aliexpress.com/item/1005005268679505.html with a GC9D01 Controller and a resolution of 40x160 Pixel.

Kind regards
Christoph

**Comment by Basitadam** [comment]
There is no pin description in the page. Did the seller send you it? Can you send your setup.h file

**Reply by webfraggle** [reply]
I'm not using the Arduino IDE, i'm using platform.io and these are my build flags:
`build_flags = 

	-DDEBUG=1
	-DUSER_SETUP_LOADED=1
	-DGC9D01_DRIVER=1
	-DTFT_WIDTH=160
	-DTFT_HEIGHT=160
	; -DCGRAM_OFFSET=1
	-DTFT_RGB_ORDER=0
	-DTFT_RST=21 ; -> 4 RES
	-DTFT_CS=-1
	-DTFT_DC=47 ; -> 3 RS
	-DTFT_MOSI=11 ; -> 1 SDA
	-DTFT_SCLK=12 ; -> 2 SCL
	-DLOAD_GLCD=1
	-DLOAD_GFXFF=1
	-DSPI_FREQUENCY=20000000
	-DSPI_READ_FREQUENCY=20000000
	; -DSPI_TOUCH_FREQUENCY=2500000
	-DTFT_CS1=45 ; -> 5 CS
	-DTFT_CS2=48 ; -> 5 CS
	-DTFT_CS3=1
	-DTFT_CS4=2
	-DTFT_BL=14
	-DBUTTON_PIN=0
	-DBUTTON1_PIN=40
	-DBUTTON2_PIN=39
	-DNEOPIXEL_PIN=38
	-DCONFIG_SPIRAM_SUPPORT=1
`

I have the pinout from another webpage: 
![61jAJU6tmiL _AC_UF350,350_QL80_](https://github.com/user-attachments/assets/76463253-a81e-4adc-9f3f-250628cc0207)

**Comment by Basitadam** [comment]
Shouldn't one of these be 40?
-DTFT_WIDTH=160
-DTFT_HEIGHT=160

**Comment by webfraggle** [comment]
I had some Displays where I had to use a much larger size with an offset. But using the right resolution doesn't not change anything.

But I found out why your code isn't working. The controller is a GC9D01N and Liligo has a module with that display and a github repo for that.
And I used their init sequence and it works.

https://github.com/Xinyuan-LilyGO/T-0.99TFT/blob/main/src/TFT_GC9D01N.cpp

Thanks again for you work

**Reply by Basitadam** [reply]
Thank you for sharing. Did you build a set of eSPI files?

**Comment by webfraggle** [comment]
@Basitadam  I have created a fork of Bodmers repo, so I can use it easy in my projects:

https://github.com/webfraggle/TFT_eSPI

**Reply by mikerinderknecht** [reply]
hi @webfraggle did the  GC9D01N code you found work out of the box? I got this one from Aliexpress and I am having problems: https://de.aliexpress.com/item/1005005406788891.html
Seems to correspond to this product: https://www.jx-wisevision.com/099-small-size-40-rgbx160-dots-tft-lcd-display-module-screen-product/ from which I got the pinout to connect to my ESP32

**Comment by webfraggle** [comment]
@mikerinderknecht 
I'm using my fork at the moment. I can't remember what and if I really changed something. 

In my platform.ini  I'm using this branch: https://github.com/webfraggle/TFT_eSPI.git#GC9D01N
